_posts/2020-10-27-rails-upgrade-shims.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: post
+title: "Shims that you might encounter during a Rails Upgrade"
+date: 2020-10-27 10:00:00
+categories: ["rails", "upgrades"]
+author: "luciano"
+---
+
+When upgrading a Rails application, you might find that sometimes functionality is extracted from Rails and moved into a new gem. These gems are called shims, and basically allow you to keep using an old functionality, once the core API takes that out.
```

Comment:
`These gems are called shims` is that an official terminology?

Reply:
It's not a commonly used term in Rails, but here you can see Heroku using it: https://blog.heroku.com/rails-5-beta-upgrade#shim-gems

Reply:
Just added a link to give some more context https://github.com/fastruby/blog/pull/198/commits/abbb5abd9cb9be5cc090ebec1a11aff36a52ead0